search/query/date_range.go

```diff
@@ -84,19 +85,74 @@ func (t *BleveQueryTime) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func DateRangeUnmarshal(input []byte, obj *DateRangeQuery) error {
```

Comment:
Don't see a need for this method to be exported.

Also perhaps we should change the signature to this method to  ..

    dateRangeQuery, err := UnmarshalDateRange(input)

Separately can you help clarify the need for this validation here?

Reply:
ack i will not export
signature is the same for all json.Unmarshall calls in parse.go.
So i have kept it consistent

I am basically avoiding the usage of UnmarshalJSON() which is called when json.Unmarshal is called, for a date range query object in the ParseQuery path. By using a custom unmarshaller i can avoid entering the User Path here. Thus i can keep the start and end as raw strings and NOT time objects. 

When the searcher func is called, i have access to the index mapping and can get the layout to parse the raw start and end strings with.